From 48ee5fec5670ac1baae838554987f2bcf2d03b96 Mon Sep 17 00:00:00 2001
From: Sean Busbey <busbey@apache.org>
Date: Mon, 6 Jul 2015 17:00:51 -0500
Subject: [PATCH 176/226] HBASE-14027 clean up multiple netty jars.

(cherry picked from commit 37e273b8bb7b895a1e143ecf9dfdff29e03e4837)

Conflicts:
	hbase-server/pom.xml
	pom.xml

Reason: Bugfix
Author: Sean Busbey
Ref: CDH-29009
---
 hbase-it/pom.xml |    6 ++++--
 pom.xml          |    9 +++++++++
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/hbase-it/pom.xml b/hbase-it/pom.xml
index c65f52d..91d3964 100644
--- a/hbase-it/pom.xml
+++ b/hbase-it/pom.xml
@@ -212,10 +212,12 @@
       <groupId>org.apache.htrace</groupId>
       <artifactId>htrace-core</artifactId>
     </dependency>
+    <!-- Hadoop needs Netty 3.x at test scope for the minicluster -->
     <dependency>
-      <groupId>org.jboss.netty</groupId>
+      <groupId>io.netty</groupId>
       <artifactId>netty</artifactId>
-      <version>3.2.4.Final</version>
+      <version>${netty.hadoop.version}</version>
+      <scope>test</scope>
     </dependency>
   </dependencies>
 
diff --git a/pom.xml b/pom.xml
index 206afa6..cc25997 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1114,6 +1114,7 @@
     <jamon-runtime.version>2.3.1</jamon-runtime.version>
     <jettison.version>1.3.3</jettison.version>
     <netty.version>3.6.6.Final</netty.version>
+    <netty.hadoop.version>3.6.6.Final</netty.hadoop.version>
     <joni.version>2.1.2</joni.version>
     <jcodings.version>1.0.8</jcodings.version>
     <!-- Plugin Dependencies -->
@@ -1834,6 +1835,10 @@
                 <artifactId>servlet-api</artifactId>
               </exclusion>
               <exclusion>
+                <groupId>io.netty</groupId>
+                <artifactId>netty</artifactId>
+              </exclusion>
+              <exclusion>
                 <groupId>stax</groupId>
                 <artifactId>stax-api</artifactId>
               </exclusion>
@@ -1856,6 +1861,10 @@
                 <artifactId>servlet-api</artifactId>
               </exclusion>
               <exclusion>
+                <groupId>io.netty</groupId>
+                <artifactId>netty</artifactId>
+              </exclusion>
+              <exclusion>
                 <groupId>stax</groupId>
                 <artifactId>stax-api</artifactId>
               </exclusion>
-- 
1.7.0.4

